docs/docs/en/guides/develop/build/registry.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Registry
+titleTemplate: :title · Build · Develop · Guides · Tuist
+description: Optimize your Swift package resolution times by leveraging the Tuist Registry.
+---
+
+# Registry {#registry}
+
+> [!WARNING] BETA
+> This feature is currently in beta. If you encounter any issues, please report them at our <LocalizedLink href="https://community.tuist.dev/c/troubleshooting-how-to/6">community forum</LocalizedLink>.
+
+> [!IMPORTANT] REMOTE PROJECT REQUIRED
+> This feature requires a <LocalizedLink href="/server/introduction/accounts-and-projects">remote project</LocalizedLink>.
+
+As the number of dependencies grows, so does the time to resolve them. While other package managers like [CocoaPods](https://cocoapods.org/) or [npm](https://www.npmjs.com/) are centralized, Swift Package Manager is not. Because of that, SwiftPM needs to resolve dependencies by doing a deep clone of each repository, which can be time-consuming. To address this, Tuist provides an implementation of the [Package Registry](https://github.com/swiftlang/swift-package-manager/blob/main/Documentation/PackageRegistry/PackageRegistryUsage.md), so you can download only the commit you _actually need_. If the package is in the [Swift Package Index](https://swiftpackageindex.com/), it's also in the Tuist Registry.
```

Comment:
What do you think if we are explicit about the fact that the "deep clones" happen when we are doing a clean/cold resolution?

Reply:
Also, what if we mention the following:
- The registry mirrors Swift Package Index.
- It does so serving copies of versions in a global/edge storage for minimal latency.

Reply:
> What do you think if we are explicit about the fact that the "deep clones" happen when we are doing a clean/cold resolution?

It's not just clean and cold resolutions – anytime a dependency is missing or you updated them, you'd end up doing deep clone as well. I'd rather keep the phrasing around deep clones as-is instead of trying to capture all the scenarios when SwiftPM actually does a deep clone.

I updated the docs a bit for the second comment.